### ISSUE TITLE
It is NOT supporting to Null Safety

### ISSUE THREAD

**Issue by kamleshwebtech** [aloisdeniel/flutter_geocoder#68]
I have setup it in my project but getting error:

```
The library 'package:geocoder/geocoder.dart' is legacy, and should not be imported into a null safe library.
Try migrating the imported library.
```

Please update it's library and make it compatible with Null Safety. Thanks.

**Comment by AliYar-Khan** [comment]
yeah make it null safe asap

**Comment by yahu1031** [comment]
I guess you can go with #64 @AliYar-Khan @kamleshwebtech 
```yaml
dependencies:
  flutter:
    sdk: flutter
  flutter_geocoder:
    git:
      url: https://github.com/JCKodel/flutter_geocoder.git
      ref: "b0361f4"
```
[Here is code](https://github.com/aloisdeniel/flutter_geocoder/tree/b0361f49b9002d0b4ad856dce1550fd2023d54b9)

**Comment by AlexMainaMunyua** [comment]
I am getting this error when I try your method
```
pub get failed (server unavailable) -- attempting retry 4 in 8 seconds...
Git error. Command: `git rev-list --max-count=1 b0361f4`                
stdout:                                                                 
stderr: fatal: ambiguous argument 'b0361f4': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:                   
'git <command> [<revision>...] -- [<file>...]'                          
exit code: 128    
```

**Comment by yahu1031** [comment]
@AlexMainaMunyua try this [`7059f6f`](https://github.com/JCKodel/flutter_geocoder/tree/7059f6f629336be2fa71117e971b5e7d9fd091f9)

```yaml
dependencies:
  flutter:
    sdk: flutter
  flutter_geocoder:
    git:
      url: https://github.com/JCKodel/flutter_geocoder.git
      ref: "7059f6f"
```
BTW you don't need this any more, cause the package is now sounding null-safety

**Comment by Markaleth** [comment]
Do we have a null-safe variant of this that also supports iOS?

**Comment by yahu1031** [comment]
@Markaleth yes we do. Better read the docs once. 😄

**Comment by mZadorskii** [comment]
@yahu1031 would you please help with a link, Im struggling with finding  null safe geocoder versions that support iOS. thanks!

**Comment by yahu1031** [comment]
@mZadorskii the latest version is sounding null safety. Not sure what are you struggling with 😕

**Comment by yahu1031** [comment]
@kamleshwebtech or @aloisdeniel better close this issue.

**Comment by mZadorskii** [comment]
@yahu1031 My bad, I just looked yaml and saw 2.12 sdk.
pub dev is not showing Null-safe sticker next to library for some reason
![image](https://user-images.githubusercontent.com/17264517/127566716-8c1e90fa-afdb-4085-a65a-7facf04ff97b.png)

**Comment by yahu1031** [comment]
No need to worry about it buddy @mZadorskii just continue with it.

**Comment by coreysprague** [comment]
It looks like what is checked into master has been updated for null safety, but that it has not been published to pub.dev. The latest version on pub.dev is still 0.2.1 which is NOT null sound safe. Are there plans to publish 0.3.0?

**Comment by AlexMainaMunyua** [comment]
All is good now. Previously I was getting some issues.

You can close this issue.

**Comment by InfiniteTabs** [comment]
~~The linked PR that fixes the null safety seems to have an issue with converting coordinates to an address in iOS, anyone else having the same issue or knows what the issue is?   https://github.com/JCKodel/flutter_geocoder/pull/1#issuecomment-887420192~~

Just want to add I was using the git-version with the null-safety fix
```
 flutter_geocoder: 
    git:
      url: https://github.com/JCKodel/flutter_geocoder.git
      ref: "7059f6f"
```

The pub.dev is still 0.2.1 as  @coreysprague mentioned, so that is not possible to use, but I realised that using the following solves the issue.

```
 flutter_geocoder: 
    git:
      url: https://github.com/aloisdeniel/flutter_geocoder.git
      ref: "2ee33a9"
```

So basically, until the version is upgraded in pub.dev we could use the latter.

**Comment by MennaCC** [comment]
> flutter_geocoder: 
>     git:
>       url: https://github.com/aloisdeniel/flutter_geocoder.git
>       ref: "2ee33a9"

The plugin `geocoder` uses a deprecated version of the Android embedding.
